osquery/tables/system/windows/logon_sessions.cpp

```diff
@@ -0,0 +1,142 @@
+  case 13:
+    return L"CachedUnlock";
+  default:
+    return L"None";
+  }
+}
+
+QueryData QueryLogonSessions(QueryContext& context) {
+  QueryData results;
+  bool loadedManually = false;
+
+  HMODULE module = GetModuleHandle(TEXT("Secur32.dll"));
+
+  if (!module) {
+    module = LoadLibrary(TEXT("Secur32.dll"));
+    loadedManually = true;
+  }
+
+  NTSTATUS(__stdcall * LsaEnumerateLogonSessions)
+  (PULONG LogonSessionCount, PLUID * LogonSessionList);
+  LsaEnumerateLogonSessions =
+      reinterpret_cast<decltype(LsaEnumerateLogonSessions)>(
+          GetProcAddress(module, "LsaEnumerateLogonSessions"));
+
+  NTSTATUS(__stdcall * LsaGetLogonSessionData)
+  (PLUID LogonId, PSECURITY_LOGON_SESSION_DATA * ppLogonSessionData);
+  LsaGetLogonSessionData = reinterpret_cast<decltype(LsaGetLogonSessionData)>(
+      GetProcAddress(module, "LsaGetLogonSessionData"));
+
+  NTSTATUS(__stdcall * LsaFreeReturnBuffer)(PVOID Buffer);
+  LsaFreeReturnBuffer = reinterpret_cast<decltype(LsaFreeReturnBuffer)>(
+      GetProcAddress(module, "LsaFreeReturnBuffer"));
+
+  if (LsaEnumerateLogonSessions && LsaGetLogonSessionData &&
+      LsaFreeReturnBuffer) {
+    PLUID sessions = NULL;
+    ULONG sessionCount = 0;
+    NTSTATUS status = LsaEnumerateLogonSessions(&sessionCount, &sessions);
+
+    if (status == 0) {
+      for (size_t i = 0; i < sessionCount; i++) {
+        PSECURITY_LOGON_SESSION_DATA sessionData = NULL;
+        NTSTATUS status = LsaGetLogonSessionData(&sessions[i], &sessionData);
+
+        if (status == 0) {
+          Row r;
+          r["logon_id"] = INTEGER(sessionData->LogonId.LowPart);
+          r["user"] = wstringToString(sessionData->UserName.Buffer);
+          r["logon_domain"] = wstringToString(sessionData->LogonDomain.Buffer);
+          r["authentication_package"] =
+              wstringToString(sessionData->AuthenticationPackage.Buffer);
+          r["logon_type"] =
+              wstringToString(GetLogonType(sessionData->LogonType));
+          r["session_id"] = INTEGER(sessionData->Session);
+          LPTSTR sid;
+          if (ConvertSidToStringSid(sessionData->Sid, &sid)) {
+            r["logon_sid"] = sid;
+          }
+          if (sid) {
+            LocalFree(sid);
+          }
+          r["logon_time"] = BIGINT(filetimeToUnixtime(sessionData->LogoffTime));
+          r["logon_server"] = wstringToString(sessionData->LogonServer.Buffer);
+          r["dns_domain_name"] =
+              wstringToString(sessionData->DnsDomainName.Buffer);
+          r["upn"] = wstringToString(sessionData->Upn.Buffer);
+          r["logon_script"] = wstringToString(sessionData->LogonScript.Buffer);
+          r["profile_path"] = wstringToString(sessionData->ProfilePath.Buffer);
+          r["home_directory"] =
+              wstringToString(sessionData->HomeDirectory.Buffer);
+          r["home_directory_drive"] =
+              wstringToString(sessionData->HomeDirectoryDrive.Buffer);
+          results.push_back(r);
```

Comment:
Cool, the new update has been updated to reflect this. It is probably worth reviewing most of the tables if this is not the most efficient method since most of them do it like my initial code.